### PR TITLE
feat: add Dockerfile and docker-compose.yml

### PR DESCRIPTION
**Pull request by Yochyo**
The PR contains a Dockerfile and sample docker-compose.yml file to run this project inside a container.
This is the first time I've done anything like that, so some things could probably be improved.

arguments can be configured using environment variables. The environment variables are the following:
```
API_KEY: a simple string.
API_URL: (optional) simple string
QUERY: (optional) tag query string
THRESHOLD:  (optional) decimal number (for example 75.0)

CERT: (optional) boolean. If true, the certificate must be mounted at "/usr/src/app/cert"
OVERWRITE: (optional) boolean, sets -overwrite flag
SKIP_HASHING: (optional) boolean, sets -skip-hashing flag
CLEAR_SEARCH_CACHE: (optional) boolean, sets -clear-search-cache flag
VERBOSE: (optional) boolean, sets -verbose flag
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM ubuntu:latest
+WORKDIR /usr/src/app
+RUN apt-get update
```
Comment (appleappleapplenanner) [inline Dockerfile:3]: Combine apt-get update and apt-get install into one layer. [See this](https://docs.docker.com/develop/dev-best-practices/) and [this](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) for explanation.

**Review by appleappleapplenanner — CHANGES_REQUESTED** [review]
Good work, thanks for doing this.

I added a couple comments:
- Use Python 3.11 for performance
- Query should be JSON now and doesn't need to be a CLI parameter
- Set API_URL to reach the Docker host by default
- Other random nitpicky comments

I attached the files that implement my suggested changes. If you commit those I'll merge.

[changes.zip](https://github.com/appleappleapplenanner/hydrus-video-deduplicator/files/11951683/changes.zip)

**Comment by Yochyo** [comment]
Thank you a lot for taking the time to review it and especially for making suggestions and explaining why you changed something.

This PR should now be ready for merging now.

**Comment by appleappleapplenanner** [comment]
Good work

**Pull request merged**